functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,222 @@
+package functional_tests
+
+import (
+	"context"
+	"go/build"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	gstruct "github.com/onsi/gomega/gstruct"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	machinev1 "github.com/openshift/client-go/machine/clientset/versioned"
+	. "github.com/project-codeflare/codeflare-common/support"
+	"github.com/project-codeflare/instascale/controllers"
+	"github.com/project-codeflare/instascale/pkg/config"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	mc "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/client/clientset/versioned"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client // You'll be using this client in your tests.
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+	err       error
+)
+
+func startEnvTest(t *testing.T) *envtest.Environment {
+	test := With(t)
+	//specify testEnv configuration
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "config", "crd", "bases"),
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "project-codeflare", "multi-cluster-app-dispatcher@v1.38.1", "config", "crd", "bases"),
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api@v0.0.0-20220411210816-c3bb724c282a", "machine", "v1beta1"),
+		},
```

Comment:
This could also be added to throw an error if there's an issue with the CRD path:

```go
ErrorIfCRDPathMissing: true,
```